### PR TITLE
CSAF 2.1 CSD02 Public Feedback Resolution and Editor Revision (for TC meeting 2026-04-29)

### LINKED ISSUE #1311: JSONPath
We should revisit where it makes sense to switch to JSON Path instead of JSON Pointer.

### LINKED ISSUE #1316: Remove Notes and What is new
I suggest to remove the Notes and What is new file for CSAF 2.1.

Reasoning: Both are far from being up-to-date and I think no one has the intention (or time/resources) to update them.

@sthagen / @santosomar : Thoughts?

### LINKED ISSUE #1356: Minor: missing comma in 9.1.14
There is a missing comma between the links to "3" and "6.1" in 9.1.14:
> satisfies those normative requirements in sections [2.4](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#extensions), [3](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#schema-elements) [6.1](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#mandatory-tests), [6.4](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#test-presets), and [8](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#safety-security-and-data-protection-considerations) that are designated as applying to CSAF Validators.

Diff:
> satisfies those normative requirements in sections [2.4](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#extensions), [3](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#schema-elements), [6.1](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#mandatory-tests), [6.4](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#test-presets), and [8](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#safety-security-and-data-protection-considerations) that are designated as applying to CSAF Validators.

Issue was announced here: https://groups.oasis-open.org/discussion/new-issue-minor-missing-comma-in-9114

### LINKED ISSUE #1357: Missing dates in 6.1.37 list
Test 6.1.37 describes that it's validation steps need to be run for every item of type `string` and format `date-time`.

In the list of relevant paths for test 6.1.37, I believe the paths to EPSS and SSVC timestamps are missing.

These would be:
`/vulnerabilities[]/metrics[]/content/epss/timestamp`
`/vulnerabilities[]/metrics[]/content/ssvc_v2/timestamp`

Does the fact that the `ssvc_v2` property is in a different schema matter here? It is defined as type `string` and format `date-time` at [https://certcc.github.io/SSVC/data/schema/v2/SelectionList_2_0_0.schema.json](https://certcc.github.io/SSVC/data/schema/v2/SelectionList_2_0_0.schema.json)

This issue was announced here: https://groups.oasis-open.org/discussion/new-issue-missing-dates-in-6137-list

### LINKED ISSUE #1364: Clarification for license expressions 6.1.54
Concerning test 6.1.54: 
It is not fully clear what is meant by 
> check for the ABNF defined in annex B of [SPDX301].

The ABNF defines `license-id = <short form license identifier from SPDX License List>`, but do we have to validate that the used `license-id` is indeed in the list of SPDX licenses, or is it enough to check that it could syntactically be a SPDX License (i.e., `idstring` that does not start with `DocumentRef-`, `LicenseRef-` or `AdditionRef-`)?
The fact that Test **6.2.45 Non-Existing License Identifier** explicitly tests that `license-id` and
ScanCode LicenseRefs exist (and the example contains a non-LicenseRef expression) indicates that test 6.1.54 should not yet fail for non-existing license ids.

Announced: https://groups.oasis-open.org/discussion/new-issue-clarification-for-license-expressions-6154

### LINKED ISSUE #1365: Grammar fix 6.3.18 Use of Qualitative Severity Rating
Currently:

`This covers all items in metrics regardless of their origin. Even though the Qualitative Severity Rating is a specified property, **it's** usage is discouraged as it provides no information about the assessment inputs. `

Fix:

`This covers all items in metrics regardless of their origin. Even though the Qualitative Severity Rating is a specified property, **its** usage is discouraged as it provides no information about the assessment inputs. 
`

Announced:
https://groups.oasis-open.org/discussion/new-issue-grammar-fix-6318-use-of-qualitative-severity-rating

### LINKED ISSUE #1367: [CWE-A] link malformed
In 6.11 it seems there is an extra ] and this breaks the internal link
``A list of all CWE release archives is available at [[CWE-A]](#CWE-A])].``

Announced:
https://groups.oasis-open.org/discussion/new-issue-cwe-a-link-malformed

### LINKED ISSUE #1368: Add RVISC governance requirements
RVISC is referenced in several places in CSAF 2.1 incl mandatory tests, recommended tests, converter requirements etc but there is currently no defined governance for the registry itself.

I request the TC to consider a normative reference to a RVISC governance document maintained by the TC, updatable independently covering registration criteria and overall process. This keeps the spec stable, while we still ensure to the registrants and implementers how this registry operates. 

https://groups.oasis-open.org/discussion/new-issue-add-rvisc-governance-requirements

### LINKED ISSUE #1369: Add additional test cases for Mandatory Test 6.1.3 (Circular Definition of Product ID)
With the schema changes in CSAF 2.1, the product_paths structure has become more complex compared to CSAF 2.0 relationships. The increased complexity (nested subpaths, beginning_product_reference, full_product_name.product_id) introduces more edge cases for circular dependency detection that are not yet covered by the existing test case (03-01).

Falling Cases:
1. Direct self-cycle: subpath next_product_reference points back to beginning_product_reference of the same path
```
"product_paths": [
      {
        "beginning_product_reference": "CSAFPID-0001",
        "full_product_name": { "name": "Product A", "product_id": "CSAFPID-0002" },
        "subpaths": [
          { "category": "installed_on", "next_product_reference": "CSAFPID-0001" }
        ]
      }
    ]
```

2. Inter-path cycle: A → B → C → A
```
"product_paths": [
      {
        "beginning_product_reference": "CSAFPID-0001",
        "full_product_name": { "name": "Product B", "product_id": "CSAFPID-0002" },
        "subpaths": [
          { "category": "installed_on", "next_product_reference": "CSAFPID-0003" }
        ]
      },
      {
        "beginning_product_reference": "CSAFPID-0002",
        "full_product_name": { "name": "Product C", "product_id": "CSAFPID-0003" },
        "subpaths": [
          { "category": "installed_on", "next_product_reference": "CSAFPID-0001" }
        ]
      }
    ]
```

Valid Case:
1. Diamond-shaped graph (A → B → D, A → C → D): the same node D is reachable via two paths but no cycle exists. This specifically tests the backtracking step to prevent false positives.
```
 "product_paths": [
  {
    "beginning_product_reference": "CSAFPID-0001",
    "full_product_name": { "name": "Product B", "product_id": "CSAFPID-0002" },
    "subpaths": [
      { "category": "installed_on", "next_product_reference": "CSAFPID-0004" }
    ]
  },
  {
    "beginning_product_reference": "CSAFPID-0001",
    "full_product_name": { "name": "Product C", "product_id": "CSAFPID-0003" },
    "subpaths": [
      { "category": "installed_on", "next_product_reference": "CSAFPID-0004" }
    ]
  }
] 
```

### LINKED ISSUE #1370: 9.1.18: remove model_numbers, serial_numbers and skus conversion rules that always adds a `*`
From 2.1 Committee Specification Draft 02 9.1.18:

> * type `/$defs/full_product_name_t/product_identification_helper/model_numbers[]`:
>      * If a model number is given that does not end on a star, the CSAF 2.0 to CSAF 2.1 Converter SHOULD add a `*` to the end and output a  warning that a partial model number was detected and a star has been added.

looks like a defect in the rule, as 3.1.4.3.3 has:

>  Any given model number of value type `string` with at least `1` character represents a model number of the component to identify - possibly with placeholders.

so it is okay to have a model_number without placeholders. I assume it specifies a full model_number.
And in the examples there are model numbers that only have `?` placeholders.
(CSAF 2.0 also allows full model numbers and partial ones that do not end with a `*`.)

I suggest to remove that subitem. (Or provide a reason why all model_numbers should end with a `*`.)

## same situation with serial_numbers and skus.

The same issue is there with `serial_numbers` and `skus`.

## subsequent removal rule

There is the `skus` example `2.063.*.?:221-A`, that shows what would happen, if a `*` was added, as the next rule is

> If the stock keeping unit contains multiple unescaped * after the conversion, the CSAF 2.0 to CSAF 2.1 Converter MUST remove the entry 

So all entries will be removed. 🤔 🤯 

## combine all three rules
... to shorten the text and avoid feedback like this from a reporter that goes through the rules steps by steps and works on them. 😁

### LINKED ISSUE #1371: 9.1.18: minor markup ambiguity TLP:<ValidTLPLabel>
https://github.com/oasis-tcs/csaf/blob/17f90fd3078feaf05109f016ec3bda9d696100b2/csaf_2.1/prose/share/csaf-v2.1-draft.md?plain=1#L14391

is probably meant to put the `<ValidTLPLabel>` outside of the Markdown code quote, to make sure it is not matched literally.

Suggestion:

```md
If `/document/distribution/text` contains the string `TLP v2.0: TLP:` followed by a valid label, the CSAF 2.0 to CSAF 2.1 Converter SHOULD provide an option to use it instead.
```

which also avoids introducing `<ValidTLPLabel>`

### LINKED ISSUE #1372: 9.1.5 + 9.1.18: clarify "output an information" in converter rules
There is one place in each converter requirements that says to "output an information":

https://github.com/oasis-tcs/csaf/blob/17f90fd3078feaf05109f016ec3bda9d696100b2/csaf_2.1/prose/share/csaf-v2.1-draft.md?plain=1#L13729-L13730

https://github.com/oasis-tcs/csaf/blob/17f90fd3078feaf05109f016ec3bda9d696100b2/csaf_2.1/prose/share/csaf-v2.1-draft.md?plain=1#L14464-L14465

A warning seems to be better here, because
 * it seems like many other warning situations in the converters
 * if it is the only "information" instead of a "warning", it complicates implementations that would need to treat it differently from a warning, just for one case.

BTW: The phrase `The output MUST include the original values and, if applicable, the converted ones.` should be deleted as there is a general requirement now from the general requirement of #1343.

"output" is not clear here as well as the resulting new CSAF 2.1 file will have the new values of course (that is one type of output) and the "warning" or "information" will have the original values (as the other type of output)

### LINKED ISSUE #1373: 9.1.18 Lose false friend "loose"
In 9.1.18 "losing information" is meant - I think, so replace three times:

`s/loos/los/g` in the document

References:
* https://en.wiktionary.org/wiki/lose
*  https://en.wiktionary.org/wiki/loose
* 
https://github.com/oasis-tcs/csaf/blob/17f90fd3078feaf05109f016ec3bda9d696100b2/csaf_2.1/prose/share/csaf-v2.1-draft.md?plain=1#L14490-L14495

### LINKED ISSUE #1374: 6.2.38: Clarify handling of unicode variants
In https://github.com/oasis-tcs/csaf/issues/1250, the need for clarification of how to handle different unicode variants when handling `/document/category` for 6.1.26 was recognized.

I would argue that the same applies to 6.2.38, so please clarify if when checking for `csaf_deprecated_`:
* the `/document/category` value should to be lowercased
* unicode underscore variants need to be considered for the two underscores in `csaf_deprecated_`
* underscores, hyphens (and variants thereof) and whitespaces need to be considered when their are the only prefix of `csaf_deprecated_`, i.e. if ` csaf_deprecated_security_advisory`, `_csaf_deprecated_security_advisory`, `-csaf_deprecated_security_advisory` should fail the test

Edit: This issue was announced here: https://groups.oasis-open.org/discussion/new-issue-6238-clarify-handling-of-unicode-variants

### LINKED ISSUE #1405: Editorial
- [x] Conversion rule for CPE missing in CVRF CSAF converter
- [x] 9.1.18 remove double rule regarding "path with warning"

### LINKED ISSUE #1411: [Editorial] Conformance changes
> I suggest we do three things:
> 
> 1. Fix the MUST vs. SHALL divergence in conformance section

I guess we should also make sure, we insert MUST/SHALL where we currently write just plain statements about the programs...

> 2. Remove superficial / non-semantic language like `output warning` with `issue warning`

We should also consider, whether the phrasing is ok for errors. My experiments with the "stochastic parrots" additionally ended up in either "display" (which I think is not the right word if you put stuff in a log file) or "output" :see_no_evil: 

> 3. Split the conformance source file along the second level section boundaries

:heavy_plus_sign:

_Originally posted by @tschmidtb51 in https://github.com/oasis-tcs/csaf/issues/1402#issuecomment-4331047319_

### LINKED ISSUE #1432: CVSS 4.0
We need to fix the links or clarify that the schema is available at `https://www.first.org/cvss/cvss-v4.0.json`  but the schema id is `https://www.first.org/cvss/cvss-v4.0.2.json` - or convince @nickleali to add a symlink from `https://www.first.org/cvss/cvss-v4.0.2.json` to `https://www.first.org/cvss/cvss-v4.0.json` (or the other way around...; preferred 😄 )

### LINKED ISSUE #1435: Nit: `translation_json_schema.json` id / title / description mismatch
From the `translations_json_schema.json` [file](https://github.com/oasis-tcs/csaf/blob/3c0e44a2d2753efbbe26e4e02b18313b2668f7ff/csaf_2.1/test/language_specific_translation/translations_json_schema.json)

```
  "$id": "https://raw.githubusercontent.com/oasis-tcs/csaf/master/csaf_2.1/test/validator/testcases_json_schema.json",
  "title": "Test cases for CSAF",
  "description": "Representation of the data provided for test cases from section 6 of the specification.",
```

The files $id, title and description do not match the content of the file.

Announced here: https://groups.oasis-open.org/discussion/new-issue-nit-translation-json-schemajson-id-title-description-mismatch

### LINKED ISSUE #1449: makefile: set SHELL
The `SHELL` variable defaults to `/bin/sh` if not set in the makefile. Under Linux, it is not taken from the environment, so we should set it manually (cf. https://www.gnu.org/software/make/manual/html_node/Choosing-the-Shell.html)
The target `render-pdf` uses `pushd` which is not available in `/bin/sh`.

### PR DESCRIPTION
**Pull request by tschmidtb51**
# Content Changes

- [x] addresses parts of #1396
- [x] fixes #1372 (#1404)
- [x] fixes #1405 (#1412, #1440, #1448)
- [x] fixes #1368 (#1427)
- [x] fixes #1432 (#1438, #1448)
- [x] fixes #1357 (#1440)

# Editorial Changes

- [x] fixes #1365 (#1398)
- [x] fixes #1370 (#1401)
- [x] fixes #1371 (#1402)
- [x] fixes #1374 (#1406)
- [x] fixes #1364 (#1407)
- [x] fixes #1367 (#1408)
- [x] fixes #1316 (#1409)
- [x] fixes #1373 (#1429)
- [x] fixes #1411 (#1436)
- [x] fixes #1435 (#1437)
- [x] fixes #1356 (#1440)
- [x] fixes #1311 (#1442) 
- [x] Fixed [\# 1421 - simplified revision history section](https://github.com/oasis-tcs/csaf/issues/1421) (#1451)

# Maintenance Changes

- [x] Maint: asterisk and dash cleansing [\# 1453](https://github.com/oasis-tcs/csaf/pull/1453)

# Tool Changes

- [x] fixes #1449 (#1450)

# Test Files Changes

- [x] fixes #1369 (#1422, #1447)

### CODE REVIEW

**Review by sthagen — APPROVED** [review]

**Review by sthagen — APPROVED** [review]

**Comment by tschmidtb51** [comment]
Todo:

- [x] update comment resolution log based on https://github.com/oasis-tcs/csaf/issues/1396#issuecomment-4347088763
- [x] update comment resolution log based on https://github.com/oasis-tcs/csaf/issues/1362#issuecomment-4347114695
- [x] add missing documentation from PRs merged
- [x] discuss #1382
- [x] render delivery items

**Review by sthagen — APPROVED** [review]

**Review by sthagen — APPROVED** [review]

**Review by sthagen — APPROVED** [review]

**Review by sthagen — APPROVED** [review]

**Review by sthagen — APPROVED** [review]

**Comment by tschmidtb51** [comment]
Motion set and seconded: https://groups.oasis-open.org/discussion/motion-for-1399

> I hereby submit the following motion and request that if seconded and no objection received per this list before one week has passed on 2026-05-22 12:00 UTC to automatically carry.
The Chair usually states the result per mail to this list when the period has passed.

**Review by sthagen — APPROVED** [review]

**Comment by sthagen** [comment]
Incoming change set in:
- "Maint: asterisk and dash cleansing" [\# 1453](https://github.com/oasis-tcs/csaf/pull/1453) has been included.

If we would have forgotten, we could have applied before the next editor revision easily ...

PS: inside `csaf_2.1/prose/edit` ... do 1.:
```bash
if [ "$(uname)" = 'Darwin' ]; then quux="''"; else quux=''; fi;
for f in src/*.md; do \
  sed -i ${quux} \
  "s/^\*\ /- /g; s/^\ \ \*\ /  - /g; s/^\ \ \ \ \*\ /    - /g; s/^\ \ \ \ \ \ \*\ /      - /g; " \
  $f; done
for f in $(grep "^-------" src/*.md | cut -f1 -d:); do \
  sed -i '' "s/^-------/---/g;" \
  $f; done
```

and finally 2. edit `etc/markdownlint.json`:
```diff
diff --git a/csaf_2.1/prose/edit/etc/markdownlint.json b/csaf_2.1/prose/edit/etc/markdownlint.json
index eeb14631..e504b8a3 100644
--- a/csaf_2.1/prose/edit/etc/markdownlint.json
+++ b/csaf_2.1/prose/edit/etc/markdownlint.json
@@ -14,7 +14,7 @@
     "style": "atx"
   },
   "hr-style": {
-    "style": "-------"
+    "style": "---"
   },
   "line-length": false,
   "link-fragments": false,
@@ -55,6 +55,6 @@
     "style": "asterisk"
   },
   "ul-style": {
-    "style": "asterisk"
+    "style": "dash"
   }
 }
```

**Review by justmurphy — APPROVED** [review]
lgtm

**Comment by justmurphy** [comment]
Motion passed 5/22 and merged on 5/27 during TC meeting: https://groups.oasis-open.org/discussion/motion-for-1399
